internal/zinc-persist/src/main/scala/sbt/internal/inc/binary/converters/ProtobufReaders.scala

```diff
@@ -0,0 +1,641 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package sbt.internal.inc.binary.converters
+
+import java.io.File
+
+import sbt.internal.inc.Relations.ClassDependencies
+import sbt.internal.inc._
+import sbt.util.InterfaceUtil
+import xsbti.{ Position, Problem, Severity, T2, UseScope }
+import xsbti.compile.{ CompileOrder, FileHash, MiniOptions, MiniSetup, Output, OutputGroup }
+import xsbti.compile.analysis.{ Compilation, ReadMapper, SourceInfo, Stamp }
+import sbt.internal.inc.binary.converters.ProtobufDefaults.Feedback.StringToException
+import sbt.internal.inc.binary.converters.ProtobufDefaults.Feedback.{ Readers => ReadersFeedback }
+import sbt.internal.inc.binary.converters.ProtobufDefaults.{ Classes, ReadersConstants }
+import sbt.internal.util.Relation
+import xsbti.api._
+
+final class ProtobufReaders(mapper: ReadMapper) {
+  def fromPathString(path: String): File = {
+    java.nio.file.Paths.get(path).toFile
+  }
+
+  def fromStampType(stampType: schema.Stamps.StampType): Stamp = {
```

Comment:
It's a good idea, but the whole readers/writers are internal. The reason I decided to leave them like that is just in case some downstream user needs them at any point... That way Zinc wouldn't be blocking her/him from using them. What do you think?